src/AutoRest.CSharp/Mgmt/Report/TransformableItem.cs

```diff
@@ -18,20 +18,28 @@ public TransformableItem(string fullSerializedName, TransformSection transformSe
         }
 
         public string FullSerializedName { get; set; }
-        protected virtual List<string>? TransformTypeWhiteList { get { return null; } }
+        protected virtual HashSet<string>? TransformTypeWhiteList { get { return null; } }
+
+        private List<string>? _appliedTransformLogs;
 
         [YamlMember(DefaultValuesHandling = DefaultValuesHandling.OmitEmptyCollections | DefaultValuesHandling.OmitNull)]
         [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
         public List<string>? TransformLogs
         {
             get
             {
-                var r = _transformSection.GetAppliedTransformLogs(
-                    this.FullSerializedName, this.TransformTypeWhiteList)
-                    .OrderBy(item => item.Log.Index)
-                    .Select(item => $"[{item.Log.Index}][{item.Transfom}] {item.Log.LogMessage}").ToList();
-                // return null when it's an empty list so that it will be ignored in Json
-                return r.Count == 0? null : r;
+                {
+                    if (_appliedTransformLogs is null)
+                    {
+                        var r = _transformSection.GetAppliedTransformLogs(
+                        this.FullSerializedName, this.TransformTypeWhiteList)
+                        .OrderBy(item => item.Log.Index)
+                        .Select(item => $"[{item.Log.Index}][{item.Transform}] {item.Log.LogMessage}").ToList();
+                        // return null when it's an empty list so that it will be ignored in Json
+                        _appliedTransformLogs = r.Count == 0 ? null : r;
```

Comment:
Going purely by the comment here, do we want to cache the result as empty and return null instead when the expression is indeed empty? That way you would cache empty value too.

Reply:
Updated the code to set the value of _appliedTransformLogs value immediately after running GetAppliedTransformLogs once and doing the null check in the return statement.

Reply:
Not sure how this change helps? You are still assigning _appliedTransformLogs to null when r is empty so you will continue to reevaluate it every time r.count is empty - no?

Reply:
I think you are looking at the outdated code.